code/obj/machinery/flasher.dm

```diff
@@ -227,14 +218,57 @@
 
 	return
 
+/obj/machinery/flasher/portable //Portable version of the flasher. Only flashes when anchored
+	name = "portable flasher"
+	desc = "A portable flashing device. Wrench to activate and deactivate. Cannot detect slow movements."
+	icon_state = "pflash1-c"
+	strength = 8
+	anchored = 0
+	base_state = "pflash"
+	density = 1
+	event_handler_flags = USE_PROXIMITY | USE_FLUID_ENTER
+	var/cooldown_scan = 1.5 SECONDS
+	var/cooldown_end = 0
+
+/obj/machinery/flasher/portable/power_change()
+	..()
+	if (powered())
+		if (!src.anchored)
+			icon_state = "[base_state]1-c"
+		else
+			if (GET_COOLDOWN(src, "flash"))
+				icon_state = "[base_state]1-c"
+			light.enable()
+
 /obj/machinery/flasher/portable/HasProximity(atom/movable/AM as mob|obj)
-	if ((src.disable) || (src.last_flash && world.time < src.last_flash + 150))
+	if (!src.anchored || src.disable)
+		return
+
+	if (GET_COOLDOWN(src, "flash"))
+		return
+
+	if (!powered())
 		return
 
-	if(iscarbon(AM))
+	if (iscarbon(AM))
 		var/mob/living/carbon/M = AM
-		if ((M.m_intent != "walk") && (src.anchored))
-			src.flash()
+		if (M.m_intent != "walk")
+			if (ishuman(M))
```

Comment:
I adjusted the code to just be on var/mob/ but it doesn't actually make a difference because they are failing the iscarbon check. No mob critters trigger the flasher, this is identical behavior to master right now.

Reply:
Changed to catch the living stuff